include/DetectorHit.h

```diff
@@ -21,27 +21,32 @@ class DetectorHit : public G4VHit
     DetectorHit(); //< standard constructor
     virtual ~DetectorHit();
     void SetEdep(G4double de)  { fEdep = de; }
-    void SetTime(G4double val) {fTime = val;}
+    void SetTime(G4double val, G4double weight ) {fTime = weight*val;}
```

Comment:
I see a inconsistency in this approach of weighing time and position with energy - in this setter weight can be put as an argument - it could be whatever, not necessarily fEdep. But getters return always this value divided by fEdep. That could lead to some errors, that would not be easy to recognize. 
The solution would be to initialize fEdep (and other variables), use it as the weight, when method setEnergy would be called, recalculate time and position using the new weight.

Reply:
> Move all methods' definitions to *.cpp file.

I prefer not. In header they will be inlined, so we will gain on execution time